### PR TITLE
Feature/change SetStartingLpBasis

### PR DESCRIPTION
**Pull request by sgatto**
We should remove the `SetStartingLpBasisInt` function in order to be more consistent with the wishes of google/ortools.
This pr aims to delete `SetStartingLpBasisInt` and `GetStartingLpBasisInt`

### CODE REVIEW

File: ortools/linear_solver/xpress_interface.cc [hunk 1]
```diff
@@ -1749,17 +1751,16 @@ void XpressInterface::SetStartingLpBasis(
...
-  std::vector<int> initCstat;
-  std::vector<int> initRstat;
+  initial_variables_basis_status_.clear();
+  initial_variables_basis_status_.reserve(variable_statuses.size());
   for (const MPSolver::BasisStatus& status : variable_statuses) {
-    initCstat.push_back(MPSolverToXpressBasisStatus(status));
+    initial_variables_basis_status_.push_back(MPSolverToXpressBasisStatus(status));
   }
+  initial_constraint_basis_status_.reserve(constraint_statuses.size());
+  initial_constraint_basis_status_.clear();
```
Comment (flomnes) [inline ortools/linear_solver/xpress_interface.cc:1760]: `clear` then `reserve`

File: ortools/linear_solver/xpress_interface.cc [hunk 2]
```diff
@@ -1243,7 +1242,7 @@ int64_t XpressInterface::nodes() const {
 }
 
 // Transform a XPRESS basis status to an MPSolver basis status.
-MPSolver::BasisStatus XpressInterface::xformBasisStatus(
+MPSolver::BasisStatus XpressInterface::XpressToMPSolverBasisStatus(
```
Comment (flomnes) [inline ortools/linear_solver/xpress_interface.cc:1245]: Move out of class
```suggestion
MPSolver::BasisStatus XpressToMPSolverBasisStatus(
```

File: ortools/linear_solver/xpress_interface.cc [hunk 3]
```diff
@@ -422,7 +419,7 @@ class XpressInterface : public MPSolverInterface {
   void AddSolutionHintToOptimizer();
 
   // Transform XPRESS basis status to MPSolver basis status.
-  static MPSolver::BasisStatus xformBasisStatus(int xpress_basis_status);
+  static MPSolver::BasisStatus XpressToMPSolverBasisStatus(int xpress_basis_status);
```
Comment (flomnes) [inline ortools/linear_solver/xpress_interface.cc:422]: Move out of class

File: ortools/linear_solver/xpress_interface.cc [hunk 4]
```diff
@@ -2091,7 +2076,7 @@ void XpressInterface::Write(const std::string& filename) {
   VLOG(1) << "Writing Xpress MPS \"" << filename << "\".";
   const int status = XPRSwriteprob(mLp, filename.c_str(), "");
   if (status) {
-    LOG(ERROR) << "Xpress: Failed to write MPS! ";
+    LOG(ERROR) << "Xpress: Failed to write MPS! ExtractModel";
```
Comment (flomnes) [inline ortools/linear_solver/xpress_interface.cc:2079]: ?

Comment (sgatto) [reply ortools/linear_solver/xpress_interface.cc:2079]: ehm, no idea

File: ortools/linear_solver/xpress_interface.cc (hunk 3 above)
Comment (sgatto) [reply ortools/linear_solver/xpress_interface.cc]: done

File: ortools/linear_solver/xpress_interface.cc (hunk 2 above)
Comment (sgatto) [reply ortools/linear_solver/xpress_interface.cc]: done

File: ortools/linear_solver/xpress_interface.cc [hunk 5]
```diff
@@ -1745,38 +1742,23 @@ void XpressInterface::SetLpAlgorithm(int value) {
...
-    return;
+  initial_variables_basis_status_.clear();
+  initial_variables_basis_status_.reserve(variable_statuses.size());
+  for (const MPSolver::BasisStatus& status : variable_statuses) {
+    initial_variables_basis_status_.push_back(MPSolverToXpressBasisStatus(status));
+  }
+  initial_constraint_basis_status_.clear();
+  initial_constraint_basis_status_.reserve(constraint_statuses.size());
+  for (const MPSolver::BasisStatus& status : constraint_statuses) {
+    initial_constraint_basis_status_.push_back(MPSolverToXpressBasisStatus(status));
```
[the remark this answers is not in the document]
Comment (sgatto) [reply ortools/linear_solver/xpress_interface.cc]: done

File: ortools/linear_solver/xpress_interface_test.cc [hunk 6]
```diff
@@ -1307,7 +1354,6 @@ TEST(XpressInterface, CallbackThrowsException) {
 }  // namespace operations_research
 
 int main(int argc, char** argv) {
-  InitGoogle(argv[0], &argc, &argv, true);
```
Comment (flomnes) [inline ortools/linear_solver/xpress_interface_test.cc:1310]: Why remove this ?

Comment (sgatto) [reply ortools/linear_solver/xpress_interface_test.cc:1310]: It brakes the standard googletest arguments apparently without adding functionalities that are useful for us

**Review by flomnes — APPROVED** [review]

**Pull request merged**